### PR TITLE
Move prow CRD yaml to one level below

### PR DESCRIPTION
**Pull request by chaodaiG**
This will make it possible for make rules to do kubectl apply -f config/prow/cluster, without need to know which prow CRD is applied. For now we'll use schemaless CRD

### CODE REVIEW

**Comment by chaodaiG** [comment]
Related https://github.com/kubernetes/test-infra/pull/24148#issuecomment-990121197

File: config/prow/Makefile [hunk 1]
```diff
@@ -32,6 +32,7 @@ update-plugins: get-cluster-credentials
 	kubectl create configmap plugins --from-file=plugins.yaml=plugins.yaml --dry-run -o yaml | kubectl replace configmap plugins -f -
 
 deploy-prow: get-cluster-credentials
+	kubectl apply --server-side=true -f ./cluster/prowjob-crd/legacy/prowjob-schemaless_customresourcedefinition.yaml
```
Comment (cjwagner) [inline config/prow/Makefile:35]: Are there any issues with using server-side apply for the schemaless CRD?

**Review by cjwagner — COMMENTED** [review]
Please also update [`deploy.sh`](https://github.com/kubernetes/test-infra/blob/07d01555bb5110f6b69f011a6579430c0a2a8170/prow/deploy.sh#L89)

File: config/prow/Makefile (hunk 1 above)
Comment (chaodaiG) [reply config/prow/Makefile:35]: should not be

**Comment by chaodaiG** [comment]
> Please also update [`deploy.sh`](https://github.com/kubernetes/test-infra/blob/07d01555bb5110f6b69f011a6579430c0a2a8170/prow/deploy.sh#L89)

nice catch, done!

**Review by cjwagner — APPROVED** [review]

**Pull request merged**